sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -753,6 +753,8 @@ object SparkSession {
 
     private[this] val options = new scala.collection.mutable.HashMap[String, String]
 
+    private[this] var master: Option[String] = None
```

Comment:
`master` is only useful when creating a new `SparkContext`, so here I separate it from `options` and only use it when creating new `SparkContext`